fix(css): explain flex-basis 0 v 0% (#35549)
mdn/content: files/en-us/web/css/flex/index.md

- `<'flex-shrink'>`
  - : Defines the {{cssxref("flex-shrink")}} of the flex item. Negative values are considered invalid. Defaults to `1` when omitted. (initial is `1`)
- `<'flex-basis'>`

>   - : Defines the {{cssxref("flex-basis")}} of the flex item. A preferred size of `0` must have a unit to avoid being interpreted as a flexibility. Defaults to `0%` when omitted. The initial value is `auto`.

Reviewer: ```suggestion
  - : Defines the {{cssxref("flex-basis")}} of the flex item. A preferred size of `0` must have a length unit to avoid being interpreted as a flexible value. Defaults to `0%` when omitted. The initial value is `auto`.
```

Reviewer: not sure if this suggestion is correct or what you intended. "Flexibility" is vague.

Author: I am not sure what the original author intended here by ```A preferred size of `0` must have a length unit...```. The `flex: 1 1 0` and `flex: 1 1 0em` both serialize to `flex-basis: 0px`. We could remove the statement:
```suggestion
- : Defines the {{cssxref("flex-basis")}} of the flex item. Defaults to `0%` when omitted. The initial value is `auto`.
```

- `none`: The flex item neither grows nor shrinks. This value expands to `flex: 0 0 auto`. The item is sized according to its `width` or `height` properties, depending on the direction of the flex container. The flex item is fully inflexible.
- `flex: <number [1,∞]>`: The flex item's main size will be proportional to the number set. This value expands to `flex: <number> 1 0`. This sets the `flex-basis` to zero and makes the flex item flexible. The item will be at least as wide or tall as its minimum size, with the container's positive available space being proportionally distributed based on the growth factors of this item and its sibling flex items. If all the flex items use this pattern, all will be sized in proportion to their numeric values.

  > [!WARNING]

>   > The browsers use `flex-basis` value `0%` when the `flex-basis` is not specified in a `flex` value. This is not the same as `flex-basis` value `0` which is what the specification says. This may affect flex layout in some cases.

Reviewer: ```suggestion
  > The browsers use `flex-basis` value `0%` when the `flex-basis` is not specified in a `flex` value. This is not the same as `flex-basis` value `0` which is what the specification says. This may affect flex layout in some cases. See this effect demonstrated in the [Flex-basis `0` versus `0%`](en-US/docs/Web/CSS/flex-basis#flex_basis_0_vs_0) example.
```
